packages/meditrak-app/app/utilities/userLocation/actions.js

```diff
@@ -46,6 +47,16 @@ export const watchUserLocation = () => async (dispatch, getState) => {
   });
 };
 
+export const refreshWatchingUserLocation = () => dispatch => {
+  dispatch({ type: CLEAR_LOCATION_DATA });
+
+  Geolocation.getCurrentPosition(
+    position => dispatch(onGeolocationPosition(position.coords)),
+    error => dispatch(onGeolocationError(error.message)),
+    GEOLOCATION_OPTIONS,
+  );
+};
```

Comment:
Is this necessary? The watch should still continue to update things when the location changes, I would have thought.

If the watch doesn't continue updating, then we also need to add a new watch here, similar to in `watchUserLocation`, so that the coordinates automatically update as you wander around

Reply:
You are totally right @edmofro, I must admit I have a very shallow understanding of what is going on here.